Pull forDebuggingOnly to main explainer. (#1019)
WICG/turtledove: FLEDGE.md

*   forDebuggingOnly *win* report from `generateBid()` of the buyer that wins the whole auction (both the component auction and top level auction).
*   forDebuggingOnly *win* report from `scoreAd()` of the component seller for the buyer that wins the whole auction.
*   forDebuggingOnly *win* report from `scoreAd()` of the top level seller for the winning component seller (who is on behalf of its winning buyer).
*   forDebuggingOnly *loss* report from all other `generateBid()` and `scoreAd()`.

##### 7.1.1 Post Auction Signals   

A post auction signal is a signal which is only available after the auction completes, such as the highest scoring other bid. The forDebuggingOnly APIs support the text placeholders below, which will be replaced with the corresponding value from the auction when found in the reporting URL's query parameters.

*   “${winningBid}” - The value of the winning bid. In component auctions, this value comes from the component auction and not the top-level auction.
*   “${winningBidCurrency}” - If the auction has a `sellerCurrency` configured, this will be its currency tag; otherwise it is `'???'` to denote that it's in the bidder's original currency.

> *   “${madeWinningBid}” - A boolean value representing whether the owner of this interest group made the winning bid, either via this interest group, or another interest group with the same owner. In component auctions, this value comes from the component auction and not the top-level auction.

Reviewer: ```suggestion
*   “${madeWinningBid}” - A [Boolean](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) value representing whether the owner of this interest group made the winning bid, either via this interest group, or another interest group with the same owner. In component auctions, this value comes from the component auction and not the top-level auction.
```

*   “${highestScoringOtherBid}” - The value of the bid that was scored as second highest by the seller’s scoreAd script. Note that this may not be the second highest bid value, since scores and bids may be independent. This value comes from and is only reported in component auctions but not top-level auctions, and is not reported to losing bidders.
*   “${highestScoringOtherBidCurrency}” - The currency `highestScoringOtherBid` is in. If the auction has a `sellerCurrency` configured, this will be its currency tag; otherwise it is `'???'` to denote that it's in the bidder's original currency.

> *   “${madeHighestScoringOtherBid}” - A boolean value representing whether the owner of this interest group made the ${highestScoringOtherBid} bid, either via this interest group, or another interest group with the same owner. This value comes from and is only reported in component auctions but not top-level auctions, and is not reported to losing bidders. If there’s a tie for ${highestScoringOtherBid} from more than one owner, this is false for all.

Reviewer: ```suggestion
*   “${madeHighestScoringOtherBid}” - A [Boolean](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) value representing whether the owner of this interest group made the ${highestScoringOtherBid} bid, either via this interest group, or another interest group with the same owner. This value comes from and is only reported in component auctions but not top-level auctions, and is not reported to losing bidders. If there’s a tie for ${highestScoringOtherBid} from more than one owner, this is false for all.
```

*   “${topLevelWinningBid}” - The value of the bid that won the top-level auction. This value is only reported to component sellers.
*   “${topLevelWinningBidCurrency}” - The currency `topLevelWinningBid` is in.
If the top-level auction has a `sellerCurrency` configured, this will be its currency tag; otherwise it is `'???'` to denote that it's in whatever currency the component auction made the bid in.

> * `${rejectReason}` - The seller can optionally add a `rejectReason` field to its `scoreAd()` return object to convey to the bidder a more detailed reason why the bid was rejected. A component auction's bidders only get reject reasons from its component seller, but not reject reasons from the top-level seller. The reject reason returned by `scoreAd()` should be one of:

Reviewer: Should this say "must" instead of "should"?
```suggestion
* `${rejectReason}` - The seller can optionally add a `rejectReason` field to its `scoreAd()` return object to convey to the bidder a more detailed reason why the bid was rejected. A component auction's bidders only get reject reasons from its component seller, but not reject reasons from the top-level seller. The reject reason returned by `scoreAd()` must be one of:
```

    *   "not-available", which is the default value, when a bid
        *    was not rejected
        *    was rejected but the seller didn’t provide a reject reason
        *    the reason provided wasn't one of the above values
    *   "invalid-bid"
    *   "bid-below-auction-floor"
    *   "pending-approval-by-exchange"
    *   "disapproved-by-exchange"
    *   "blocked-by-publisher"
    *   "language-exclusions"
    *   "category-exclusions"

#### 7.2 navigator.deprecatedReplaceInURN()

> To help with ease of adoption, [until at least 2026](https://github.com/WICG/turtledove/issues/286#issuecomment-1682842636) we will support the API `navigator.deprecatedReplaceInURN()`.

Reviewer: ```suggestion
To help with ease of adoption, [until at least 2026](https://github.com/WICG/turtledove/issues/286#issuecomment-1682842636) Protected Audience will support the `navigator.deprecatedReplaceInURN()` API.
```
